back/src/dals/room/repository/room.repository.ts

```diff
@@ -1,5 +1,5 @@
 import { RoomInfo, RoomContext } from 'dals'
-import { InputMessageTypes } from 'messages';
+import { InputMessageTypes } from 'pods/messages';
```

Comment:
I think this import is not necessary, isn't it?

Reply:
Which one do you mean? All the imports in that file are in use

Reply:
I mean we don't need `import { InputMessageTypes } from 'pods/messages';` in dals repository, isn't it?

Reply:
The POD uses the repository, but the repository should never know about the pod, is independant, we should isolate it

Reply:
This code could lead to issues

```typescript
const updateRoomContent = async (newRoomInfo: RoomInfo, action:string): Promise<void> => {
  const room: RoomInfo = await getRoomByName(newRoomInfo.room);
  
  let newContent = "";
  
  if(room) {
    newContent = room.content;
  }
  
  switch (action) {
    case InputMessageTypes.TRAINER_APPEND_TEXT:
      newContent += '\n' + newRoomInfo.content;
      break;
    case InputMessageTypes.TRAINER_SET_FULL_TEXT:
      newContent = newRoomInfo.content;
      break;
  };
  await RoomContext.updateOne({room: newRoomInfo.room}, {content: newContent});
};
```
Rather refactor:

Option A (updated as well some variable names).

```diff
- const updateRoomContent = async (newRoomInfo: RoomInfo, action:string): Promise<void> => {
+ const updateRoomContent = async (newRoomInfo: RoomInfo, newContent :string, replaceFullText: boolean = false): Promise<void> => {
  const room: RoomInfo = await getRoomByName(newRoomInfo.room);
  
  let finalContent = "";
  
  if(room) {
    finalContent = room.content;
  }
  
-  switch (action) {
-    case InputMessageTypes.TRAINER_APPEND_TEXT:
-      newContent += '\n' + newRoomInfo.content;
-      break;
-    case InputMessageTypes.TRAINER_SET_FULL_TEXT:
-      newContent = newRoomInfo.content;
-      break;
-  };

+ finalContent = (replaceFullText) ? 
  newContent
:
`${content}\n${newContent}`;
;

  await RoomContext.updateOne({room: newRoomInfo.room}, {content: finalContent});
};
```

Option B (create two endpoints).

```typescript
const appendContentToRoom = async (newRoomInfo: RoomInfo, content: string): Promise<void> 
const replaceFullTextContentInRoom = async (newRoomInfo: RoomInfo, content: string): Promise<void> ;
```

I know all this likely is something inherited from the past and not related to the changes you have introduced but we are following the "good boy scout rule" here, when we find something we check if we can get it better if it's related with the PR (if not a separate case is opened), in this case is related with the PR (refactor)

Reply:
I have just pushed some changes to solve the problem.

Hopefully it works for you.